### PR TITLE
Add Classic Bluetooth link support

### PR DESCRIPTION
**Pull request by zxzxwu**
I want to test some features which can only work on Classic Bluetooth, such as role switching(in the following commits). However, we cannot rely on RootCanal because it's not mature as well.

The first commit centralize Local Linked Devices usage to a shared util module so that the following change won't change a lot. Then, I rename the original `LocalLink` class to `LocalLeLink`, and add a `LocalClassicLink` class to handle classic transport.

For the Remote Link path, I am not sure what we should do for it.

### CODE REVIEW

File: bumble/controller.py [hunk 1]
```diff
@@ -432,16 +481,19 @@ def on_link_peripheral_disconnected(self, peer_address):
...
             )
 
-    def on_link_acl_data(self, sender_address, data):
+    def on_link_acl_data(self, sender_address, transport, data):
         # Look for the connection to which this data belongs
-        connection = self.find_connection_by_address(sender_address)
+        if transport == BT_LE_TRANSPORT:
+            connection = self.find_le_connection_by_address(sender_address)
+        else:
+            connection = self.find_classic_connection_by_handle(sender_address)
```
Comment (barbibulle) [inline bumble/controller.py:496]: this looks odd, calling `find_classic_connection_by_handle` with an address instead of a handle.

Comment (zxzxwu) [reply bumble/controller.py]: Done. Making a2dp test on BR/EDR to verify this.

File: bumble/controller.py [hunk 2]
```diff
@@ -82,23 +94,33 @@ def on_hci_acl_data_packet(self, packet):
...
         self.hci_sink = None
         self.link = link
 
-        self.central_connections = (
-            {}
-        )  # Connections where this controller is the central
-        self.peripheral_connections = (
-            {}
-        )  # Connections where this controller is the peripheral
+        self.central_connections: dict[
```
Comment (uael) [inline bumble/controller.py:115]: nit: Use `Dict` instead of `dict`

**Review by uael — APPROVED** [review]

File: bumble/controller.py (hunk 2 above)
Comment (zxzxwu) [reply bumble/controller.py]: Done

**Review by barbibulle — APPROVED** [review]

**Review by uael — APPROVED** [review]

**Pull request merged**